### DISCUSSION TITLE
Solweig: No valid DEM selected

### DISCUSSION THREAD

**Discussion by elidamd** [UMEP-dev/UMEP#387] [General]
Hi!

I am running SOLWEIG without success. I would like to ask for some tips on how to find out the error (what is wrong with the files). Thanks!

1. Before [plugin version 3.11] 
SOLWEIG was successful with DSM, landcover, sky view factor inputs. 
However, it was returning empty when I added vegetation DSM files (from tree generator tool). 
2. Now [updated plugin to 3.19.3] 
I am trying to run SOLWEIG with the same inputs from before, but it gives the error: "No valid DEM selected". 
My dsm had negative values, so I adjusted (raster calculator)
It is still giving me the error, and no other message that I could understand what is the problem with the file.
To check the files, I was able to generate the "daily shadow pattern" function (using both CDSM and DSM I am applying Solweig).

 - OS: Windows 10
 - QGIS Version: 3.22.2
 - UMEP Version: 3.19.3
 - SuPy Version: 2022.2.17

Thanks a lot!

**Comment by biglimp** [comment]
SOLWEIG need to know the locations of buildings. This is established by either:

- comparing a DSM (ground and building heights) with a DEM (ground heights) OR
- retrieving that information from a land cover dataset

Your error implies that you have no DEM loaded.

**Reply by elidamd** [reply]
Oh, I thought it was an error in the file. Thank you very much!
